pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1031,11 +1031,17 @@ func buildUpstreamClusterTLSContext(opts *buildClusterOpts, tls *networking.Clie
 			Sni:              tls.Sni,
 		}
 
-		// If tls.CaCertificate or CaCertificate in Metadata isn't configured don't set up SdsSecretConfig
-		if res.GetRootResourceName() == "" {
+		if res.GetRootResourceName() == "" && features.EnableDefaultToSystemCAValidation.Get() {
+			res = model.SdsCertificateConfig{
+				CaCertificatePath: features.SystemCAFilePath.Get(),
```

Comment:
You lost the SNI ? 

Also if SNI is empty, we need to default to hostname.

Reply:
SNI is actually set before all this,  check where the tlscontext is initialized in this function

Reply:
https://github.com/istio/istio/blob/4e7294303ac89e71c4d718220162f616d3a57867/pilot/pkg/networking/core/v1alpha3/cluster.go#L1008

Reply:
https://github.com/istio/istio/blob/4e7294303ac89e71c4d718220162f616d3a57867/pilot/pkg/networking/core/v1alpha3/cluster.go#L1010